scripts/alert_emails/generate_daily_alerts.ts

```diff
@@ -0,0 +1,120 @@
+/**
+ * Generates the daily alert file. Given two snapshots, finds differences in the
+ * overall threat level and each metric and generates a summary file.
+ *
+ * Run via: yarn generate-daily-alerts [old_snapshot new_snapshot]
+ *
+ * If no snapshots provided, uses the master snapshot and the current branch's
+ * snapshot.
+ *
+ * Generates a file alerts/<old_snapshot>-<new_snapshot>.json containing an
+ * array of alerts. See the Alert interface below for schema.
+ */
+
+import fs from 'fs-extra';
+import path from 'path';
+import {
+  fetchAllStateProjections,
+  fetchAllCountyProjections,
+} from '../../src/common/utils/model';
+import { ProjectionsSet } from '../../src/common/models/ProjectionsSet';
+import { Projections } from '../../src/common/models/Projections';
+import { fetchMasterSnapshotNumber, snapshotFromUrl, snapshotUrl } from '../../src/common/utils/model';
+import { SNAPSHOT_URL } from '../../src/api';
+import { Level } from '../../src/common/level';
+import { Metric, ALL_METRICS } from '../../src/common/metric';
+
+const outputFolder = path.join(__dirname, 'alerts');
+
+// Disable counties; mostly useful for faster testing / debugging.
+const STATES_ONLY = false;
+
+interface Alert {
+  fips: string;
+  oldLevel: Level;
+  newLevel: Level;
+  changedMetrics: Array<{
+    metric: Metric,
+    oldLevel: Level,
+    newLevel: Level
+  }>;
+}
+
+async function main() {
+  const {oldSnap, newSnap} = await parseArgs();
+
+  console.log('Fetching projections from snapshots...');
+  const projectionsSet = ProjectionsSet.fromProjections(
+    await fetchAllProjections(snapshotUrl(oldSnap)),
+    await fetchAllProjections(snapshotUrl(newSnap))
+  );
+
+  const alerts: Alert[] = [ ];
+  for(const pair of projectionsSet.pairs) {
+    const oldLevel = pair.left.getAlarmLevel();
+    const newLevel = pair.left.getAlarmLevel();
+    const changedMetrics = [];
+    for(const metric of ALL_METRICS) {
+      if (metric !== Metric.FUTURE_PROJECTIONS) {
+        const oldLevel = pair.left.getMetricLevel(metric);
+        const newLevel = pair.right.getMetricLevel(metric);
+        if (oldLevel !== newLevel) {
+          changedMetrics.push({ metric, oldLevel, newLevel });
+        }
+      }
+    }
+    if (oldLevel !== newLevel || changedMetrics.length > 0) {
+      alerts.push({
+        fips: pair.fips,
+        oldLevel,
+        newLevel,
+        changedMetrics
+      });
+    }
+  }
+
+  await fs.ensureDir(outputFolder);
+  const file = path.join(outputFolder, `${oldSnap}-${newSnap}.json`);
+  await fs.writeFile(file, JSON.stringify(alerts));
```

Comment:
why did you do Alert[] vs {[fips: string]: Alert}. easier only the 'finding people to email side' to have it be in an object-- do we expect there for any reason to be duplicates

Reply:
1. It seems intuitive to me to just generate a list of alerts to send.
2. There will not be duplicates, but I don't think we need to enforce that by using a different data structure.
3. Why is an object easier?  Array seems simpler since you can just loop over the values?

Reply:
Switched this over to a map so it plays better with your script. 😁